src/toxcore/dht/codec.rs

```diff
@@ -132,9 +134,13 @@ impl Decoder for DhtCodec {
         }
 
         match Packet::from_bytes(buf) {
-            IResult::Incomplete(needed) => Err(DecodeError::incomplete_packet(needed, buf.to_vec())),
-            IResult::Error(error) => Err(DecodeError::deserialize(error, buf.to_vec())),
-            IResult::Done(_, packet) => {
+            Err(Err::Incomplete(needed)) => Err(DecodeError::incomplete_packet(needed, buf.to_vec())),
+            Err(Err::Error(error)) => {
+                let (_, kind) = error;
+                Err(DecodeError::deserialize(kind, buf.to_vec()))
+            },
+            Err(Err::Failure(e)) => panic!("Packet deserialize failed with unrecoverable error: {:?}", e),
```

Comment:
There mustn't be panics on packets decoding, otherwise it'll be easy to kill the node.

Reply:
@NamsooCho please fix

Reply:
done

Reply:
Maybe must change String to Vec<u8>.

Reply:
Still there is `panic!` macro in your code. Please fix

Reply:
https://github.com/tox-rs/tox/pull/374/files#diff-24a8beb3c46ed160ba513a45d8432efaR594